### PR TITLE
[aws-lambda] Add AutoScalingScaleInHandler

### PR DESCRIPTION
**Pull request by elliot-nelson**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://docs.aws.amazon.com/autoscaling/ec2/userguide/lambda-custom-termination-policy.html
- ~~If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.~~

> Updated the existing `cognito` tests to fix an existing build break.

### CODE REVIEW

**Review by jamesorlakin — COMMENTED** [review]
Thanks for the PR! Just a couple of small tweaks and I think it'll be good to go.

(I'll be honest and say I've no idea why CI failed initially for the unrelated Cognito tests. I can't reproduce the issue on f29cca2 locally)

File: types/aws-lambda/trigger/autoscaling.d.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { Handler, Callback } from "../handler";
+
+export type AutoScalingScaleInHandler = Handler<AutoScalingScaleInEvent, AutoScalingScaleInResult>;
+export type AutoScalingScaleInCallback = Callback<AutoScalingScaleInResult>;
+
+export type AutoScalingInstanceMarketOption = "spot" | "on-demand";
```
[the remark this answers is not in the document]
Comment (elliot-nelson) [reply types/aws-lambda/trigger/autoscaling.d.ts:6]: Based on my testing, the docs are wrong -- the value AWS actually sends is "on-demand".

(I've opened https://github.com/awsdocs/amazon-ec2-auto-scaling-user-guide/pull/29 to correct the doc.)

**Review by jamesorlakin — APPROVED** [review]
Thanks @elliot-nelson

**Review by rbuckton — APPROVED** [review]

**Comment by elliot-nelson** [comment]
Ready to merge

**Pull request merged**